src/odata-to-abstract-sql.ts

```diff
@@ -823,56 +826,87 @@ export class OData2AbstractSQL {
 					}
 				}
 			}
-			if (isBindReference(key)) {
-				const bind = this.Bind(key);
-				const referencedField: ReferencedFieldNode = [
-					'ReferencedField',
-					resource.tableAlias,
-					resource.idField,
-				];
-				return [comparison.eq, referencedField, bind];
+			return this.BaseKey(resource, key);
+		}
+	}
+	BaseKey(
+		resource: AliasedResource,
+		key: NonNullable<ODataQuery['key']>,
+		followedForeignKey?: string,
+	): BooleanTypeNodes {
+		if (isBindReference(key)) {
+			const bind = this.Bind(key);
+			const referencedField: ReferencedFieldNode = [
+				'ReferencedField',
+				resource.tableAlias,
+				resource.idField,
+			];
+			return [comparison.eq, referencedField, bind];
+		}
+		const fieldNames = Object.keys(key);
+		const sqlFieldNames = fieldNames.map(odataNameToSqlName).sort();
+		// When following a FK, we check the whether the FK + the fields in the alternate notation
+		// complete a unique index. This allows users to not include the navigated FK property in
+		// the provided alternate key notation and define just the rest fields of the unique index.
+		// We effectively consider a navigated FK as implicitly defined in the alternate key notation.
+		const altSqlFieldNames =
+			followedForeignKey != null && !sqlFieldNames.includes(followedForeignKey)
+				? [...sqlFieldNames, followedForeignKey].sort()
```

Comment:
I suspect we won't need to differentiate with/without the followed FK if we aren't allowing that `/pilot?$orderby=identification_method(pilot=1,identification_type='passport')/...` case, since the followed FK should be a necessary part of the uniqueness and hence checking if it's still valid without it doesn't make sense imo

Reply:
Not necessarily. If the user is using a different natural key in the alternate notation that doesn't include the navigated FK, then that should be fine, especially after the #160. wdyt?

Reply:
did this & blocked using primitive ids as well